internal/mode/static/manager.go

```diff
@@ -265,22 +265,29 @@ func registerControllers(
 		{
 			objectType: &gatewayv1.GatewayClass{},
 			options: []controller.Option{
-				controller.WithK8sPredicate(predicate.GatewayClassPredicate{ControllerName: cfg.GatewayCtlrName}),
+				controller.WithK8sPredicate(k8spredicate.And(
+					k8spredicate.GenerationChangedPredicate{},
+					predicate.GatewayClassPredicate{ControllerName: cfg.GatewayCtlrName})),
 			},
 		},
 		{
 			objectType: &gatewayv1.Gateway{},
 			options: func() []controller.Option {
+				options := []controller.Option{
+					controller.WithK8sPredicate(k8spredicate.GenerationChangedPredicate{}),
+				}
 				if cfg.GatewayNsName != nil {
-					return []controller.Option{
-						controller.WithNamespacedNameFilter(filter.CreateSingleResourceFilter(*cfg.GatewayNsName)),
-					}
+					options = append(options,
```

Comment:
nit: we prefer to split long lines up like this:
```
options = append(
    options,
    controller.WithNamespacedNameFilter(filter.CreateSingleResourceFilter(*cfg.GatewayNsName)),
)
    
```

Reply:
Updated https://github.com/nginxinc/nginx-gateway-fabric/pull/1422/commits/de143ef24458dfeccd4a6c62746ac407f406ac78